build.rs

```diff
@@ -57,6 +57,29 @@ fn main() {
     cm.define("SDL_SHARED", "ON");
     cm.define("SDL_STATIC", "ON");
     cm.define("HIDAPI", "ON");
+
+    if cfg!(target_os = "macos") {
```

Comment:
This is super unintuitive, but makes sense if you spend enough time in `build.rs` scripts.
`cfg!(target_os = "...")` is referring to the target OS of the *`build.rs` script*, and then later when we read `env::var("CARGO_CFG_TARGET_OS")`, that's the target OS of the crate being built. `cfg!()` reads variables at compile time (of `build.rs`), while `env::var()` reads them from whenever it's run.

So this is all skipped if you try to compile for mac/ios on a not-mac platform. Which is rare but something that I think is easy enough to support. I may want to re-do this logic to accompany that... or at least put that disclaimer in a comment. x.x

Reply:
ohhhhhhhh

Reply:
yes please update that and comment why it's using env and not cfg

probably this is why cross-build always seems broken

Reply:
I tidied this up, but it also seems like something `cmake-rs` could figure out. I'm curious why they do not....